oci/layout/oci_src.go

```diff
@@ -71,13 +76,15 @@ func (s *ociImageSource) GetManifest(instanceDigest *digest.Digest) ([]byte, str
 		mimeType = s.descriptor.MediaType
 	} else {
 		dig = *instanceDigest
-		// XXX: instanceDigest means that we don't immediately have the context of what
-		//      mediaType the manifest has. In OCI this means that we don't know
-		//      what reference it came from, so we just *assume* that its
-		//      MediaTypeImageManifest.
-		// FIXME: We should actually be able to look up the manifest in the index,
-		// and see the MIME type there.
-		mimeType = imgspecv1.MediaTypeImageManifest
+		for _, md := range s.index.Manifests {
+			if md.Digest == dig {
```

Comment:
I don't think I understand your meaning here.  Do you mean that we should be building an index mapping tag names to digests when we load the index?

Reply:
This was just pointing out another place which expects the layout described in the “What is the overall layout is this designed to create?” comment.

{Because,As long as} the layout with duplicate descriptors for instance manifests is intentional, this code makes sense.